### PR TITLE
internal/extsvc/bitbucketserver: Wrap json unmarshal error

### PR DESCRIPTION
**Pull request by indradhanush**
The bare bones error message is confusing. Wrapping it to provide more context to the user.. Originally reported in [Slack](https://sourcegraph.slack.com/archives/C04M5M8GWCX/p1678239526686919).

**Current behaviour**

image

**With the fix**

image

## Test plan

- No functional change
- Tested locally
- Builds should pass

### CODE REVIEW

File: internal/extsvc/bitbucketserver/client.go [hunk 1]
```diff
@@ -986,7 +986,11 @@ func (c *Client) do(ctx context.Context, req *http.Request, result any) (*http.R
 	if s, ok := result.(*[]byte); ok {
 		*s = bs
 	} else if result != nil {
-		return resp, json.Unmarshal(bs, result)
+		err := json.Unmarshal(bs, result)
+		if err != nil {
+			return resp, errors.Wrap(err, "failed to unmarshal response to JSON")
+		}
+		return resp, nil
```
Comment (sashaostrikov) [inline internal/extsvc/bitbucketserver/client.go:993]: you can even do:
```suggestion
		return resp, errors.Wrap(json.Unmarshal(bs, result), "failed to unmarshal response to JSON")
```

when an error from unmarshal is `nil` -- it is still `nil` after wrapping

**Review by sashaostrikov — COMMENTED** [review]
LGTM, but you can simplify code :)

File: internal/extsvc/bitbucketserver/client.go (hunk 1 above)
Comment (indradhanush) [reply internal/extsvc/bitbucketserver/client.go:993]: Aha yes! Great catch - I forgot about that.

**Review by sashaostrikov — APPROVED** [review]
:shipit:

**Pull request merged**